pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -118,6 +118,20 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	// get all ClusterRoleNames that do not exist as ClusterRole
+	clusterRoleNamesNotOnCluster := populateCrClusterRoleNames(instance, clusterRoleList)
+	if len(clusterRoleNamesNotOnCluster) != 0 {
+		// update condition if any ClusterRoleName does not exist as a ClusterRole
+		instance.Status.Conditions = controllerutil.UpdateCondition(instance.Status.Conditions, "ClusterRole for ClusterPermission does not exist", clusterRoleNamesNotOnCluster, true, managedv1alpha1.SubjectPermissionStateFailed, managedv1alpha1.ClusterRoleBindingCreated)
```

Comment:
So does this condition only get added once?

Reply:
Yes, per discussion on Slack following the "all or nothing" approach. and one condition for clusterPermission and one condition for permission